siac/rentercmd.go

```diff
@@ -294,6 +301,48 @@ func rentercontractscmd() {
 	w.Flush()
 }
 
+// rentercontractsviewcmd is the handler for the command `siac renter contracts <id>`.
+// It lists details of a specific contract.
+func rentercontractsviewcmd(cid string) {
+	var rc api.RenterContracts
+	err := getAPI("/renter/contracts", &rc)
+	if err != nil {
+		die("Could not get contract details: ", err)
+	}
+
+	for _, rc := range rc.Contracts {
+		if rc.ID.String() == cid {
+			fmt.Printf(`
+Contract %v
+Host: %v (Public Key: %v)
+
+Start Height: %v
+End Height:   %v (%v blocks remaining)
+
+Total cost:        %v (Fees: %v)
+Funds Allocated:   %v
+Upload Spending:   %v
+Storage Spending:  %v
+Download Spending: %v
+Remaining Funds:   %v
+
+File Size: %v
+`, rc.ID, rc.NetAddress, rc.HostPublicKey, rc.StartHeight, rc.EndHeight,
+				rc.EndHeight-rc.StartHeight, currencyUnits(rc.TotalCost),
+				currencyUnits(rc.Fees),
+				currencyUnits(rc.TotalCost.Sub(rc.Fees)),
+				currencyUnits(rc.UploadSpending),
+				currencyUnits(rc.StorageSpending),
+				currencyUnits(rc.DownloadSpending),
+				currencyUnits(rc.RenterFunds),
+				filesizeUnits(int64(rc.Size)))
+			return
+		}
+	}
+
+	fmt.Printf("Error: Contract with id %s could not be found.\n", cid)
```

Comment:
kind of a nitpick, but we generally don't preface errors in siac. Something like `Contract not found` or `No record of that contract` is sufficient because the user already knows what the ID is.